src/target/riscv/riscv.c

```diff
@@ -3790,6 +3791,22 @@ int riscv_openocd_poll(struct target *target)
 		LOG_TARGET_DEBUG(target, "resume all");
 		riscv_resume(target, true, 0, 0, 0, false);
 	} else if (halted && running) {
+		foreach_smp_target(entry, targets)
```

Comment:
I wonder if this loop can be skipped if `haltgroup_supported` is equal to false.